tango/local_workspace.py

```diff
@@ -574,18 +554,59 @@ def registered_run(self, name: str) -> Run:
         run_dir = self.runs_dir / name
         if not run_dir.is_dir():
             raise KeyError(name)
-        with SqliteDict(self.step_info_file, flag="r") as d:
-            steps_for_run = {}
+        steps_for_run = self._load_registered_run(name)
+        return Run(name, steps_for_run, datetime.fromtimestamp(run_dir.stat().st_ctime))
+
+    def _run_step_info_file(self, name: str):
+        return self.runs_dir / name / "stepinfo.json"
+
+    def _save_registered_run(self, name: str, all_steps: Iterable[Step]):
+        unique_ids = []
+        with SqliteDict(self.step_info_file) as d:
+            for step in all_steps:
+                try:
+                    step_info = d[step.unique_id]
+                    step_info.name = step.name if step.name != step.unique_id else None
+                    d[step.unique_id] = step_info
+                except KeyError:
+                    d[step.unique_id] = StepInfo(
+                        step.unique_id,
+                        step.name if step.name != step.unique_id else None,
+                        step.__class__.__name__,
+                        step.VERSION,
+                        {dep.unique_id for dep in step.dependencies},
+                        step.cache_results,
+                    )
+                d.commit()
```

Comment:
Don't `commit()` every time. Just do it after all steps are done. I realize the old code did this, but fewer bigger transactions are going to be faster than lots of small ones.